huaweicloud/resource_huaweicloud_bms_instance.go

```diff
@@ -0,0 +1,595 @@
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud bms client: %s", err)
+	}
+
+	createOpts := &baremetalservers.CreateOpts{
+		Name:      d.Get("name").(string),
+		ImageRef:  d.Get("image_id").(string),
+		FlavorRef: d.Get("flavor_id").(string),
+		MetaData: baremetalservers.MetaData{
+			OpSvcUserId: d.Get("op_svc_userid").(string),
+			AgencyName:  d.Get("agency_name").(string),
+		},
+		UserData:         []byte(d.Get("user_data").(string)),
+		AdminPass:        d.Get("admin_pass").(string),
+		KeyName:          d.Get("key_name").(string),
+		VpcId:            d.Get("vpc_id").(string),
+		SecurityGroups:   resourceBmsInstanceSecGroupsV1(d),
+		AvailabilityZone: d.Get("availability_zone").(string),
+		Nics:             resourceBmsInstanceNicsV1(d),
+		DataVolumes:      resourceBmsInstanceDataVolumesV1(d),
+		ExtendParam: baremetalservers.ServerExtendParam{
+			ChargingMode:        d.Get("charging_mode").(string),
+			PeriodType:          d.Get("period_unit").(string),
+			PeriodNum:           d.Get("period").(int),
+			IsAutoPay:           "true",
+			IsAutoRenew:         d.Get("auto_renew").(string),
+			EnterpriseProjectId: GetEnterpriseProjectID(d, config),
+		},
+	}
+
+	if hasFilledOpt(d, "eip_id") || hasFilledOpt(d, "iptype") {
+		var publicIp baremetalservers.PublicIp
+		if hasFilledOpt(d, "eip_id") {
+			publicIp.Id = d.Get("eip_id").(string)
+		} else {
+			publicIp.Eip = &baremetalservers.Eip{
+				IpType: d.Get("iptype").(string),
+				BandWidth: baremetalservers.BandWidth{
+					ShareType:  d.Get("sharetype").(string),
+					Size:       d.Get("bandwidth_size").(int),
+					ChargeMode: d.Get("bandwidth_charge_mode").(string),
+				},
+				ExtendParam: baremetalservers.EipExtendParam{
+					ChargingMode: d.Get("eip_charge_mode").(string),
+				},
+			}
+		}
+		createOpts.PublicIp = &publicIp
+	}
+
+	if v, ok := d.GetOk("system_disk_type"); ok {
+		volRequest := baremetalservers.RootVolume{
+			VolumeType: v.(string),
+			Size:       d.Get("system_disk_size").(int),
+		}
+		createOpts.RootVolume = &volRequest
+	}
+
+	if hasFilledOpt(d, "tags") {
+		tagRaw := d.Get("tags").(map[string]interface{})
+		taglist := utils.ExpandResourceTags(tagRaw)
+		createOpts.ServerTags = taglist
+	}
+
+	log.Printf("[DEBUG] Create Options: %#v", createOpts)
+
+	n, err := baremetalservers.CreatePrePaid(bmsClient, createOpts).ExtractOrderResponse()
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud baremetalserver: %s", err)
+	}
+	job_id := n.JobID
+
+	if err := baremetalservers.WaitForJobSuccess(bmsClient, int(d.Timeout(schema.TimeoutCreate)/time.Second), job_id); err != nil {
+		return err
+	}
+
+	entity, err := baremetalservers.GetJobEntity(bmsClient, job_id, "server_id")
+	if err != nil {
+		return err
+	}
+	server_id := entity.(string)
+
+	if server_id != "" {
+		d.SetId(server_id)
+		return resourceBmsInstanceRead(d, meta)
+	}
+
+	return fmt.Errorf("Unexpected conversion error in resourceBmsInstanceCreate.")
```

Comment:
Error message should not contains period.

Reply:
fixed